sntools/interaction_channels/o16nc_n.py

```diff
@@ -0,0 +1,116 @@
+mN = 939.7 # neutron mass (MeV)
+epsilon = 0.001  # for approximating DiracDelta distribution below
+
+# List of neutrino flavors ("e", "eb", "x", "xb") that interact in this channel.
+possible_flavors = ("e", "eb", "x", "xb")
+
+data = [[e_thr, 20.0, 25.0, 30.0, 35.0, 40.0, 45.0, 50.0, 55.0, 60.0, 65.0, 70.0, 80.0, 90.0],
+ [0.0, 0.000193, 0.0163, 0.129, 0.48, 1.22, 2.49, 4.44, 7.17, 10.7, 15.2, 20.4, 32.8, 46.8]]
+ #list of energies and cross-sections for o16nc neutron emission taken from Suzuki et al 2018.
+
+def fit():
+    """ Return a spline fit for the given emission's partial cross-section as 
+    a function of energy.
+    """
+    eNu = data[0]
+    f = interp1d(eNu, data[1], kind='cubic', 
+                 fill_value='extrapolate', bounds_error = False)
+    return f
+
+class Channel(BaseChannel):
+    def generate_event(self, eNu, dirx, diry, dirz):
+        """Return an event with the appropriate incoming/outgoing particles.
+
+        Input:
+            eNu: neutrino energy
+            dirx, diry, dirz: direction of outgoing particle (normalized to 1)
+        """
+        # Note: `self.flavor` is set during __init__
+        nu_flv = {'e': 12, 'eb': -12, 'x': 14, 'xb': -14}[self.flavor]
+
+        eE = self.get_eE(eNu, dirz)
+
+        evt = Event(2008016 if nu_flv > 0 else -2008016)
+        evt.incoming_particles.append([nu_flv, eNu, 0, 0, 1])  # incoming nu
+        evt.incoming_particles.append((8016, 14900, 0, 0, 1))  # oxygen-16 nucleus at rest
+        evt.outgoing_particles.append([2112, eE, dirx, diry, dirz])  # emitted neutron
+        # evt.outgoing_particles.append([nu_flv, eNu-e_thr, 0, 0, 1])  # outgoing nu
+        return evt
+    
+    def bounds_eE(self, eNu, *args):
+        """Return kinematic bounds for integration over eE.
+
+        Input:
+            eNu:  neutrino energy (in MeV)
+            args: [ignore this]
+        Output:
+            list with minimum & maximum allowed energy of outgoing (detected) particle
+        """
+        return [self.get_eE(eNu) - epsilon, self.get_eE(eNu) + epsilon]
+        
+    def get_eE(self, eNu, cosT=0):
+        """Return energy (in MeV) of outgoing (detected) particle.
+
+        Input:
+            eNu:  neutrino energy (in MeV)
+            cosT: cosine of the angle between neutrino and outgoing (detected) particle
+        """
+        eE = random.random()*(eNu-e_thr) + mN       
+        return eE
+    
+    def dSigma_dE(self, eNu, eE):
+        """Return differential cross section in MeV^-2.
+
+        Inputs:
+            eNu: neutrino energy
+            eE:  energy of outgoing (detected) particle
+        """
+        if eNu < e_thr or abs(self.get_eE(eNu) - eE) > epsilon:
+            # This should never happen, since we set bounds for eE and eNu accordingly above
+            # ... but just in case:
+            return 0
+
+        sigma = fit()(eNu)*10**(-42)  # cross-section at eNu from the interp1d fit 
```

Comment:
```suggestion
        sigma = fit(eNu)*10**(-42)  # cross-section at eNu from the interp1d fit 
```
Adjust here if `fit` is turned from a function into a variable above.